### PR TITLE
[Bug 776346] Checks all Cairo.ImageSurface allocations

### PR DESCRIPTION
**Pull request by yaminb**
This is a pretty quick way to avoid the crashes. 
It basically checks all allocations of Cairo.ImageSurface. If it fails, it throws an OutOfmemoryException. 
You still get an exception shows, but you just click through the dialog and can continue as normal.

Do let me know if you want this handled in more detail.

### CODE REVIEW

File: Pinta.Core/Actions/EditActions.cs [hunk 1]
```diff
@@ -263,6 +263,7 @@ private void HandlerPintaCoreActionsEditCopyActivated (object sender, EventArgs
 					return;
 			
 				ImageSurface dest = new ImageSurface (Format.Argb32, rect.Width, rect.Height);
+				Pinta.Core.Utilities.CheckHandleImageSurface (dest);
```
Comment (cameronwhite) [inline Pinta.Core/Actions/EditActions.cs:266]: It might be good to also wrap up the construction of the `ImageSurface` too, to reduce the chance of this check being missed as code shifts around. 
`CairoExtensions` is also probably a good place for this to go,
e.g. `ImageSurface dest = CairoExtensions.CreateImageSurface(Format.Argb32, rect.Width, rect.Height);`

File: Pinta.Core/Pinta.Core.csproj [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.5">
+<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.0">
```
Comment (cameronwhite) [inline Pinta.Core/Pinta.Core.csproj:2]: Could you also please revert the changes to this file?

File: Pinta.Docking/Pinta.Docking.csproj [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="4.5" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (cameronwhite) [inline Pinta.Docking/Pinta.Docking.csproj:2]: Could you also please revert the changes to this file?

File: Pinta/Pinta.csproj [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.5">
+<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.0">
```
Comment (cameronwhite) [inline Pinta/Pinta.csproj:2]: Could you also please revert the changes to this file?

**Review by cameronwhite — COMMENTED** [review]
Thanks for looking at this!
I think it looks good, just had one comment about refactoring the wrapper a bit

File: Pinta.Core/Actions/EditActions.cs (hunk 1 above)
Comment (yaminb) [reply Pinta.Core/Actions/EditActions.cs:266]: Def thought about that. I didn't want to change things up too much.  But I'll do that

File: Pinta.Core/Pinta.Core.csproj (hunk 2 above)
Comment (yaminb) [reply Pinta.Core/Pinta.Core.csproj:2]: will do

File: Pinta.Docking/Pinta.Docking.csproj (hunk 3 above)
Comment (yaminb) [reply Pinta.Docking/Pinta.Docking.csproj:2]: will do

File: Pinta/Pinta.csproj (hunk 4 above)
Comment (yaminb) [reply Pinta/Pinta.csproj:2]: will do

File: Pinta.Core/Pinta.Core.csproj (hunk 2 above)
Comment (yaminb) [reply Pinta.Core/Pinta.Core.csproj]: done

File: Pinta.Docking/Pinta.Docking.csproj (hunk 3 above)
Comment (yaminb) [reply Pinta.Docking/Pinta.Docking.csproj]: done

File: Pinta/Pinta.csproj (hunk 4 above)
Comment (yaminb) [reply Pinta/Pinta.csproj]: done

File: Pinta.Core/Actions/EditActions.cs (hunk 1 above)
Comment (yaminb) [reply Pinta.Core/Actions/EditActions.cs]: done

**Review by cameronwhite — APPROVED** [review]
Looks good, thanks!
